client/starwhale/utils/config.py

```diff
@@ -248,3 +257,42 @@ def update_instance(
         )
 
         update_swcli_config(**self._config)
+
+
+@click.group(
+    "config",
+    cls=AliasedGroup,
+    help="Configuration management, edit is supported now",
+)
+def config_cmd() -> None:
+    pass
+
+
+@config_cmd.command("edit", aliases=["e"], help="edit the configuration of swlci")
+def __edit() -> None:
+    _edit()
+
+
+def _edit() -> None:
+    _editor = ""
+    if platform == "linux" or platform == "linux2":
+        _editor = os.environ.get(SWEnv.sw_editor, "")
+        if not _editor:
+            _editor = os.environ.get("EDITOR", "vi")
+    elif platform == "darwin" or platform == "win32":
+        _editor = os.environ.get(SWEnv.sw_editor, "")
+        if not _editor:
+            raise NoSupportError(f"No {SWEnv.sw_editor} is configured")
+    else:
+        raise NoSupportError(f"Current platform is not supported by: {platform}")
+    path_config = get_swcli_config_path()
+    fd, fname = tempfile.mkstemp()
```

Comment:
we should mkstemp at the path_config dir, it will avoid cross-device link issue.
`os.rename` only works when the source and target names are on the same file system.

Reply:
tempfile removed as jialei said